### ISSUE TITLE
feature request: buttons position & behaviour

### ISSUE THREAD

**Issue by alumni** [pixel-saver/pixel-saver#13]
I would like to configure where the buttons are displayed. I personally would prefer them after the system menu, not before it. I saw the code that sets the position of the buttons, and I think you could add a setting to let the user choose either an automatic position (as it is now), or a custom left or right aligned position.

Another thing is that they only show up when a window is maximized. If a non-maximized window is focused, they will perform an action on the maximized window, not on the focused window. This is a bit tricky, since the focused application name is displayed in the top bar, but the buttons act on the other window.

My suggestion is to keep the behaviour from the original window-buttons extension: always keep them visible (this will also prevent things moving all the time in the top bar) and tie them to the currently focused window. If there are no opened windows, they could be shown as disabled (or at least should do nothing).

**Comment by beinvisible** [comment]
I'd like to agree with alumni that the behaviour to have the target-window of the buttons change (or rather not change) independently from the displayed window title is somewhat inconsistent. Although this doesn't matter much to me (i use focus follows mouse, so i focus the maximised window anyway when moving the pointer to the title/status bar), I suggest to only display the buttons if the focussed window is maximised (= only at times when the status bar actually is a window title bar).

But I'd also like to come back to the button locations: I'd much prefer to have the right window buttons as _first_ buttons right of the window title, before all the other status indicators.
The main reason for me is, that it interferes with the "workspace indicator" extension: this extension allows to change the workspace by using the mouse wheel on the indicator, but due to the window buttons of maximized windows appearing/disappearing on the right end of the status bar, the position of the indicator might annoyingly move away from the mouse pointer when scrolling through workspaces.

**Comment by sirfz** [comment]
I agree, would be great if we can change the location of the buttons (as well as have the option to not display them at all) and only have them for the currently focused app regardless of its state (maximized or not) to avoid mistakenly closing the wrong app.

**Comment by ludenticus** [comment]
So, here we have two different requests:
On the one hand, the petition to move the close button to the right of the system indicators. It makes sense to me: otherwise you have to visually search the «x»
On the other hand, the issue of having a close button exclusively related to focused windows. I agree with this request as well, because you can mistakenly close the wrong window.

**Comment by pierstitus** [comment]
I made a pull request related to this issue, https://github.com/deadalnix/pixel-saver/pull/97

I moved the buttons to just besides the AppMenu, so is is visually clear to which window the buttons belong - always the focused window. The buttons are hidden for not maximised windows, as in some cases I found it to be unclear if the buttons relate to the window or to the whole app or app main window.

**Comment by edwinhuish** [comment]
@deadalnix Could you please just add a option to remove the top right buttons?

The Window Buttons by danielkza is much profersional in this. It can move to the location as we want..

**Comment by edwinhuish** [comment]
Finally, I found a solucion at #35 

Hide the icons, and install the ``` Window Buttons by danielkza ```, now it works very well.

**Comment by bittner** [comment]
@alumni The [Gnome Tweak Tool](https://wiki.gnome.org/Apps/GnomeTweakTool) sports a "Titlebar Buttons" > "Placement" option (Left)|(Right). Does this fix the problem you want to solve?

Together with the [Hide Activities Button extension](https://extensions.gnome.org/extension/744/) and the "Top Bar" > "Activities Overview Hot Corner" option in the Gnome Tweak Tool this renders an almost Unity-like user experience in Ubuntu 17.10.
